### PR TITLE
[14.0][IMP]product_supplierinfo_for_customer_sale: Search SO by customerinfo

### PR DESCRIPTION
**Pull request by toita86**
Add a new search field that retrieves all the SOs that contains the product with the given `product_name` or `product_code`.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
LGTM

Use case: if product "Acoustic bloc screens" is sold to customer with customer name "Megablock", customer can ask you to retrieve an order containing the product "Megablock" => user needs to be able to perform search using customer name and/or customer code

With this PR we add a search field with the same properties as "Product" field in SO list view, searching for customer name and code

**Review by HekkiMelody — APPROVED** [review]
Code review, LGTM

**Comment by francesco-ooops** [comment]
@pedrobaeza good to go? :)

File: product_supplierinfo_for_customer_sale/models/sale_order_line.py [hunk 1]
```diff
@@ -11,21 +11,30 @@ class SaleOrderLine(models.Model):
 
     product_customer_code = fields.Char(
         compute="_compute_product_customer_code",
+        store=True,
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer_sale/models/sale_order_line.py]: This is a no-go. You can't convert a field to stored on a table that can be populated with a lot of records.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: product_supplierinfo_for_customer_sale/models/sale_order_line.py (hunk 1 above)
Comment (toita86) [reply product_supplierinfo_for_customer_sale/models/sale_order_line.py]: Thank you for the review, while I get your point I will like to ask how do you would you implement such a feature.

My approach derives by a similar implementation found here https://github.com/OCA/stock-logistics-workflow/blob/14.0/product_supplierinfo_for_customer_picking/models/stock_move.py .
The situation is very similar so I thought that was a possible solution. 
For backward compatibility I can also make a migration script for the `store`  attribute of the field.

But I will like to have you opinion first.

Comment (HekkiMelody) [reply product_supplierinfo_for_customer_sale/models/sale_order_line.py]: One possibility is to not store the field, and implement the `search` attribute

Comment (toita86) [reply product_supplierinfo_for_customer_sale/models/sale_order_line.py]: @aleuffre Thank you for your suggestion. I have implemented the changes based on your feedback.

**Review by francesco-ooops — APPROVED** [review]
Functional still ok!

@pedrobaeza code is good now?

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Pull request merged**